src/tests/evil/evil_test_unistd.c

```diff
@@ -25,9 +25,7 @@
 #include <io.h>
 #include <fcntl.h>
 
-# define WIN32_LEAN_AND_MEAN
-# include <winsock2.h>
-# undef WIN32_LEAN_AND_MEAN
+#include <winsock2.h>
```

Comment:
This one *must* be surrounded, tho. Or else it'll include `winsock.h` because of *reasons*.

Reply:
Are you sure? I did a bunch of tests here linking with evil, eina, eo, eolian, efl and ecore enabled and didn't noticed anything weird

Reply:
Well I guess the CI says I'm wrong
![image](https://user-images.githubusercontent.com/11988816/86370988-bd751e00-bc56-11ea-9c0f-11a04779b9ed.png)

Reply:
I think it [may include](https://gist.github.com/piscisaureus/906386#file-winsock2-h-L41) windows.h though

Reply:
It _did_ compile here, tho. I'm impressed.

Reply:
But, well, see the note: https://docs.microsoft.com/en-us/windows/win32/winsock/creating-a-basic-winsock-application

I have no idea why it works here

Reply:
> But, well, see the note: https://docs.microsoft.com/en-us/windows/win32/winsock/creating-a-basic-winsock-application
> 
> I have no idea why it works here

The note says:

> The Winsock2.h header file internally includes core elements from the Windows.h header file, so there is not usually an #include line for the Windows.h header file in Winsock applications. If an #include line is needed for the Windows.h header file, this should be preceded with the #define WIN32_LEAN_AND_MEAN macro. For historical reasons, the Windows.h header defaults to including the Winsock.h header file for Windows Sockets 1.1. The declarations in the Winsock.h header file will conflict with the declarations in the Winsock2.h header file required by Windows Sockets 2.0. The WIN32_LEAN_AND_MEAN macro prevents the Winsock.h from being included by the Windows.h header. An example illustrating this is shown below.

And for what I am understanding this is almost exactly what we are doing with this patch. Am I missing something?

Reply:
There was one time I tried to compile and yes, winsock2 tried to include winsock. But could've been a side-effect.

Reply:
There was a file including `winsock.h` instead of  `winsock2.h`, maybe..?

Reply:
Nah, it was the same point, same `#include`, switching between `winsock` and `winsock2` - one would **always** clash with the other (winsock.h included winsock2.h, winsock2.h included winsock.h).

Reply:
Okay, `winsock2.h` indeed _potentially_ includes `Windows.h`: https://github.com/tpn/winsdk-10/blob/9b69fd26ac0c7d0b83d378dba01080e93349c2ed/Include/10.0.16299.0/um/WinSock2.h#L53

Reply:
yeah, it will include `windows.h` if it is not already included, as can be seen [here](https://github.com/tpn/winsdk-10/blob/9b69fd26ac0c7d0b83d378dba01080e93349c2ed/Include/10.0.10240.0/um/Windows.h#L25). Also if `windows.h` is included by `winsock2.h` then it will not try include `winsock.h` because `winsock2.h` prevents it [here](https://github.com/tpn/winsdk-10/blob/9b69fd26ac0c7d0b83d378dba01080e93349c2ed/Include/10.0.10240.0/um/WinSock2.h#L17)). Furthermore we can see [here](https://github.com/tpn/winsdk-10/blob/9b69fd26ac0c7d0b83d378dba01080e93349c2ed/Include/10.0.10240.0/um/Windows.h#L186) what is the impact of using `WIN32_LEAN_AND_MEAN` around `windows.h`.